File: oa-backend/api/role.go
package api

import (
	"oa-backend/models"
	ginUtil "oa-backend/utils/gin"
	"oa-backend/utils/msg"
	"strconv"

	"github.com/gin-gonic/gin"
)

func AddRole(c *gin.Context) {
	var role models.Role
	_ = c.ShouldBindJSON(&role)

	code = models.GeneralInsert(&role)
	msg.Message(c, code, nil)
}

func DelRole(c *gin.Context) {
	id, err := strconv.Atoi(c.Param("id"))
	if err == nil {
		code = models.DeleteRole(id)
	} else {
		code = msg.ERROR
	}
	msg.Message(c, code, nil)
}

func EditRole(c *gin.Context) {
	var role models.Role
	_ = c.ShouldBindJSON(&role)

	code = models.UpdateRole(&role)
	msg.Message(c, code, nil)
}

func QueryRole(c *gin.Context) {
	var role models.Role
	id, err := strconv.Atoi(c.Param("id"))
	if err == nil {
		role, code = models.SelectRole(id)
	} else {
		code = msg.ERROR
	}
	msg.Message(c, code, role)
}

func QueryRoles(c *gin.Context) {
	var roleQuery models.Role
	_ = c.ShouldBindJSON(&roleQuery)

	xForms := ginUtil.GinArrayPreprocessing(c)

	xForms.Data, code = models.SelectRoles(&roleQuery, &xForms)
	msg.Message(c, code, xForms)
}

func QueryAllRole(c *gin.Context) {
	var roles []models.Role
	code = models.GeneralSelectAll(&roles, nil)
	msg.Message(c, code, roles)
}

File: oa-front/src/api/role.js
import request from './base'

export const addRole = (role) => {
    return request({
        url: '/role',
        method: 'POST',
        data: role
    })
}

export const delRole = (id) => {
    return request({
        url: '/role/' + id,
        method: 'DELETE',
    })
}

export const editRole = (role) => {
    return request({
        url: '/role',
        method: 'PUT',
        data: role
    })
}

export const queryRole = (id) => {
    return request({
        url: '/role/' + id,
        method: 'GET',
    })
}

export const queryRoles = (model, pageData) => {
    return request({
        url: '/roles',
        method: 'POST',
        data: model,
        params: {
            "pageSize": pageData.pageSize,
            "pageNo": pageData.pageNo,
        }
    })
}

export const queryAllRole = () => {
    return request({
        url: '/roles',
        method: 'GET',
    })
}